spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -823,91 +1215,30 @@ def __write_block(
-        del delayed_row_data
-
-        if block_addr > all_syn_block_sz:
-            raise Exception(
-                "Too much synaptic memory has been written: {} of {} ".format(
-                    block_addr, all_syn_block_sz))
-        if d_index is not None and index is not None and index != d_index:
-            raise Exception(
-                "Delay index {} and normal index {} do not match".format(
-                    d_index, index))
-        return block_addr, single_addr, index
+        return (row_data, delayed_row_data)
 
     def __is_direct(
-            self, single_addr, connector, pre_vertex_slice, post_vertex_slice,
-            app_edge, synapse_info):
+            self, single_addr, s_info, pre_vertex_slice, post_vertex_slice,
+            is_delayed):
         """ Determine if the given connection can be done with a "direct"\
             synaptic matrix - this must have an exactly 1 entry per row
         """
         return (
-            app_edge.n_delay_stages == 0 and
-            connector.use_direct_matrix(synapse_info) and
-            (single_addr + (pre_vertex_slice.n_atoms * BYTES_PER_WORD) <=
+            not is_delayed and
+            isinstance(s_info.connector, OneToOneConnector) and
+            isinstance(s_info.synapse_dynamics, SynapseDynamicsStatic) and
+            (single_addr + (pre_vertex_slice.n_atoms * 4) <=
                 self.__one_to_one_connection_dtcm_max_bytes) and
             (pre_vertex_slice.lo_atom == post_vertex_slice.lo_atom) and
-            (pre_vertex_slice.hi_atom == post_vertex_slice.hi_atom))
-
-    def __write_row_data(
-            self, spec, connector, pre_vertex_slice, post_vertex_slice,
-            row_length, row_data, rinfo, single_synapses,
-            synaptic_matrix_region, block_addr, single_addr, app_edge,
-            synapse_info):
-        if row_length == 1 and self.__is_direct(
-                single_addr, connector, pre_vertex_slice, post_vertex_slice,
-                app_edge, synapse_info):
-            single_rows = row_data.reshape(-1, 4)[:, 3]
-            single_synapses.append(single_rows)
-            index = self.__poptable_type.update_master_population_table(
-                single_addr, 1, rinfo.first_key_and_mask, is_single=True)
-            single_addr += len(single_rows) * BYTES_PER_WORD
-        else:
-            block_addr = self._write_padding(
-                spec, synaptic_matrix_region, block_addr)
-            spec.switch_write_focus(synaptic_matrix_region)
-            spec.write_array(row_data)
-            index = self.__poptable_type.update_master_population_table(
-                block_addr, row_length, rinfo.first_key_and_mask)
-            block_addr += len(row_data) * BYTES_PER_WORD
-        return block_addr, single_addr, index
+            (pre_vertex_slice.hi_atom == post_vertex_slice.hi_atom) and
+            not s_info.prepop_is_view and
+            not s_info.postpop_is_view)
```

Comment:
can be on line above. 

monster if. including a magic number 4.

Reply:
It is meant to be a monster if - separated this way to avoid having this in line with the code!

Fixed the 4.